Build error with "nlohmann/json.hpp: No such file or directory" in spack (unit tests build, with --test=root option)

### Check duplicate issues.

- [X] Checked for duplicates

### Description

Hi, I use Spack to evaluate Fujitsu compilers for Fugaku and FX1000.
When I tried to install this application on spack with the "--test=root" option(unit tests), I got the following error:

```
/fefs/spack2/tmp/spack-stage/spack-stage-root-6.30.02-ifueiwdgknhz2zyg3olctrljdchct3mv/spack-src/tree/dataframe/test/dataframe_datasetspec.cxx:15:10: fatal error: nlohmann/json.hpp: No such file or directory
   15 | #include <nlohmann/json.hpp>
      |          ^~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [tree/dataframe/test/CMakeFiles/dataframe_datasetspec.dir/build.make:79: tree/dataframe/test/CMakeFiles/dataframe_datasetspec.dir/dataframe_datasetspec.cxx.o] Error 1
make[2]: Leaving directory '/fefs/spack2/tmp/spack-stage/spack-stage-root-6.30.02-ifueiwdgknhz2zyg3olctrljdchct3mv/spack-build-ifueiwd'
make[1]: *** [CMakeFiles/Makefile2:51492: tree/dataframe/test/CMakeFiles/dataframe_datasetspec.dir/all] Error 2
make[1]: Leaving directory '/fefs/spack2/tmp/spack-stage/spack-stage-root-6.30.02-ifueiwdgknhz2zyg3olctrljdchct3mv/spack-build-ifueiwd'
make: *** [Makefile:169: all] Error 2
```

Attached build log.
[spack-build-out_root.txt](https://github.com/root-project/root/files/14339928/spack-build-out_root.txt)


### Reproducer

``` 
$ spack install --test=root root@6.30.02%gcc@12.2.0
```

### ROOT version

[v6-30-02](https://github.com/root-project/root/releases/tag/v6-30-02)

### Installation method

Spack package manager

### Operating system

Linux(RHEL 8.8,aarch64)

### Additional context

There is another problem when using Spack. 
Please check the issue of spack.
[Installation issue: root@6.30.02 fails to build with --test=root option #42720](https://github.com/spack/spack/issues/42720)

--- a/tree/dataframe/test/CMakeLists.txt
+++ b/tree/dataframe/test/CMakeLists.txt
@@ -45,6 +45,12 @@ if(NOT (MSVC OR (APPLE AND CMAKE_SYSTEM_PROCESSOR MATCHES arm64)) OR win_broken_
 endif()
 
 ROOT_ADD_GTEST(dataframe_datasetspec dataframe_datasetspec.cxx LIBRARIES ROOTDataFrame)
+if(builtin_nlohmannjson)
+  target_include_directories(dataframe_datasetspec PRIVATE ${CMAKE_SOURCE_DIR}/builtins)
+else()
+  target_link_libraries(dataframe_datasetspec nlohmann_json::nlohmann_json)
+endif()
+
 ROOT_ADD_GTEST(dataframe_display dataframe_display.cxx LIBRARIES ROOTDataFrame)
 ROOT_ADD_GTEST(dataframe_ranges dataframe_ranges.cxx LIBRARIES ROOTDataFrame)
 ROOT_ADD_GTEST(dataframe_leaves dataframe_leaves.cxx LIBRARIES ROOTDataFrame)

